File: sdk/web/component/kenviro/controller/LinearProgrammingController.js
var LinearProgrammingController = function LinearProgrammingController(cfg)
{
    let ctrl = {};

    ctrl.Compute = function (linearProgrammingType, inputs, weights, callback)
    {
        let args = {
            LinearProgrammingType: linearProgrammingType,
            Inputs: inputs,
            Weights: weights,
        };
        cfg.Connector.Submit(args, "ardvro/component/kenviro/LinearProgramming/Compute", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };


    return ctrl;
}

File: sdk/.net/ardvro.sdk.net/component/kenviro/controller/LinearProgrammingController.cs
namespace ardvro.sdk.net.component.kenviro.controller;

using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class LinearProgrammingController
{
    private IConnection _connection;
    public LinearProgrammingController(IConnection connection)
    {
        _connection = connection;
    }

    public void Compute (String linearProgrammingType, double[][]inputs, double[] weights, Action<object> callback)
    {
        var args = new Flexible();
        args.Add("LinearProgrammingType", linearProgrammingType);
        args.Add("Inputs", inputs);
        args.Add("Weights", weights);
        _connection.Submit(args, "ardvro/component/kenviro/LinearProgramming/Compute", callback);
    }

}
